server/model/models.go

```diff
@@ -0,0 +1,54 @@
+package model
+
+type User struct {
+	ID             int
+	githubUsername string
+	auth           string
+	role           Role
+}
+
+type Experiment struct {
+	ID          int
+	Name        string
+	Description string
+}
+
+type Assignment struct {
+	ID           int
+	UserID       int
+	PairID       int
+	ExperimentID int
+	Answer       Answer
+	Duration     int
+}
+
+type FilePairs struct {
+	ID           int
+	ExperimentID int
+	Diff         string
+	Left         File
+	Right        File
+}
+
+type File struct {
+	Name    string
+	Hash    string
+	Content string
+}
+
+type Role string
+
+const (
+	Requester Role = "requester"
+	Worker    Role = "worker"
+)
+
+type Answer string
+
+const (
+	Yes   Answer = "yes"
```

Comment:
On one hand `yes, no, maybe` sounds common enough. On another hand, I'm not 100% sure it's the best options for our tool.
I remember you spent time trying other tools. Could you please remind me what do they use as names for action?
It's very basic part of the tool, I just want to do it right. Thanks!

Reply:
That values should not be exposed in the UI, never-ever.

**TL;DR**
(this answer is based in https://github.com/src-d/code-annotation/issues/2#issuecomment-357315069 but feel free to keep answering me if the conclusions below does not satisfy you)

### MTurk

In *MTurk*, the `Assignment` tracks the `Answer` of a HIT, as we do, but the `Answer` of an `Assignment` is not defined by the platform itself but by the `Requester`.

In *MTurk* the `Answer` schema is defined by the `Project` itself,
An MTurk `Answer` schema could be whatever nested thing, a quite simple could be:
`{name:string, age:int, gender:bool, desc:string, croppedImage:[]blob}`
but we don't have that concept because we were requested:
- to use a single `Project` type, whose `Answer` schema would be: `{answer:int[1-4]}`,
- to avoid unnecessary normalization;

because of that requirements, we decide by https://github.com/src-d/code-annotation/issues/2#issuecomment-357315069 :
- to have `Experiment` instead of `Project`,
- to avoid `Answer` model, and just embed it in the `Assignment`

Considering that our accepted answers are an `enum` of size 4, we just use that "strings" instead of using `int[1-4]`.
Because of that:
- the internal value of the `Assignment.Answer` should not be exposed to the UI.
- the Front should have a mapping ( DB/API &RightArrowLeftArrow; UI ) between the `enum` values used in the DB/API, and labels that the user will see.

### Supervise.ly

Supervise.ly focuses on image labeling (so they does not have `Projects` neither) and their answers are class oriented. Because of the same reason than in MTurk, the class names are always defined by the `Requester`, and they can be whatever random word or group of chars (example:  `Car`, `Dog`, `Cake`, `UFA4E4AS`...

Reply:
yeah. It's only internal name. But I still want to have it "correct".
Looks like `yes, no, maybe, skip` are ok if we consider it as enum 🎉 
Good to go!